### PR TITLE
Fix k8s configuration for embedded journal and more

### PR DESCRIPTION
**Pull request by madanadit**
In this PR:
- Do not use hostNetworking for master
- Fix master configuration for embedded
- Remove redundant worker config
- Doc fixes

### CODE REVIEW

File: integration/kubernetes/helm-chart/alluxio/templates/config/alluxio-conf.yaml [hunk 1]
```diff
@@ -32,13 +32,13 @@ metadata:
...
+    {{if $isSingleMaster}}-Dalluxio.master.hostname={{ $fullName }}-{{ $defaultMasterName }}{{end}} -Dalluxio.master.journal.type={{ .Values.journal.type }} -Dalluxio.master.journal.folder={{ .Values.journal.folder }} {{if $isHaEmbedded}}-Dalluxio.master.embedded.journal.addresses={{range $i := until $masterCount}}{{ $fullName }}-master-{{$i}}:19200,{{end}}{{end}} {{ .Values.jvmOptions }} {{- range $key, $val := .Values.properties }} -D{{ $key }}={{ $val }} {{- end}}
```
Comment (jiacheliu3) [inline integration/kubernetes/helm-chart/alluxio/templates/config/alluxio-conf.yaml]: @madanadit Is there a scenario where the master needs to know the worker port to initiate communication?

File: docs/en/deploy/Running-Alluxio-On-Kubernetes.md [hunk 2]
```diff
@@ -305,7 +307,7 @@ create a Kubernetes Secret for the HDFS client configuration.
 ```console
 kubectl create secret generic alluxio-hdfs-config --from-file=${HADOOP_CONF_DIR}/core-site.xml --from-file=${HADOOP_CONF_DIR}/hdfs-site.xml
 ```
-These two configuration files are referred in `alluxio-master.yaml` and `alluxio-worker.yaml`.
+These two configuration files are referred in `alluxio-master.yaml` and `alluxio-worker-daemonset.yaml`.
```
Comment (jiacheliu3) [inline docs/en/deploy/Running-Alluxio-On-Kubernetes.md]: `alluxio-master.yaml` -> `alluxio-master-statefulset.yaml`

File: integration/kubernetes/helm-chart/alluxio/templates/config/alluxio-conf.yaml (hunk 1 above)
Comment (madanadit) [reply integration/kubernetes/helm-chart/alluxio/templates/config/alluxio-conf.yaml]: nope, the master does not configuration to reach the worker
if the client in the master pod needs to communicate with the worker, the master returns the worker info gathered when the worker registers (no config needed)

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Pull request merged**